lib/kubernetes/src/equality/PrometheusRule.spec.ts

```diff
@@ -0,0 +1,152 @@
+/**
+ * Copyright 2020 Opstrace, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { V1Prometheusrule } from "..";
+import { isPrometheusRuleEqual } from "./PrometheusRule";
+
+// mock logger
+jest.mock("@opstrace/utils", () => ({
+  log: {
+    debug: jest.fn
+  }
+}));
+
+type Group = NonNullable<V1Prometheusrule["spec"]["groups"]>[number];
```

Comment:
I'm a typescript newbie so bear with me here.  Where is `number` coming from? Can you add a comment pointing to information about this construct and what it's doing here?